### PR TITLE
Adding one-click-buy option to useBuyButton

### PR DESCRIPTION
**Pull request by lucis**
- Adds an `options` to `useBuyButton` hooks
- Adds a `oneClickBuy` option that will redirect the user to Checkout after clicking on the button
- For now, the Checkout URL is fixed. 

https://github.com/vtex-sites/marinbrasil.store/pull/174

### CODE REVIEW

File: packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts [hunk 1]
```diff
@@ -42,13 +50,18 @@ export const useBuyButton = (sku: Maybe<SKU>, quantity: number) => {
       setLoading(true)
       const items = [orderFormItem]
 
-      await orderForm.addToCart(items)
+      await orderForm.addToCart([orderFormItem])
```
Comment (tlgimenes) [inline packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts:53]: items array

File: packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts [hunk 2]
```diff
@@ -42,13 +50,18 @@ export const useBuyButton = (sku: Maybe<SKU>, quantity: number) => {
...
+
       sendPixelEvent({
         type: 'vtex:addToCart',
         data: {
           items,
         },
       })
+
+      if (options?.isOneClickBuy) {
+        window.location.href = '/checkout/'
```
Comment (tlgimenes) [inline packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts:63]: is it assured the pixelEvent will be processed before navigating ?

Comment (klzns) [reply packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts:63]: In Store Framework we added a spinner of 500ms, just to make sure the had a feedback and events were triggered

https://github.com/vtex-apps/add-to-cart-button/pull/44/files
https://github.com/vtex-apps/add-to-cart-button/pull/45/files

File: packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts [hunk 3]
```diff
@@ -17,7 +17,15 @@ export interface SKU {
...
 
-export const useBuyButton = (sku: Maybe<SKU>, quantity: number) => {
+type Options = {
+  isOneClickBuy?: boolean
+}
+
+export const useBuyButton = (
+  sku: Maybe<SKU>,
+  quantity: number,
+  options?: Options
```
Comment (klzns) [inline packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts:27]: Btw, in the future there will be a `seller` parameter

Comment (klzns) [reply packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts:27]: Maybe have

`sku, quantity, seller, options`

Comment (tlgimenes) [reply packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts:29]: I agree. We should have a single options parameters

File: packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts [hunk 4]
```diff
@@ -13,16 +15,38 @@ interface Seller {
...
 
 export interface SKU {
-  itemId: number
+  itemId: string
   sellers: Seller[]
 }
 
-export const useBuyButton = (sku: Maybe<SKU>, quantity: number) => {
+type Options = {
+  isOneClickBuy?: boolean
```
Comment (tlgimenes) [inline packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts:23]: why adding this `is` and not calling it just `oneClickBuy` ?

File: packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts [hunk 5]
```diff
@@ -13,16 +15,38 @@ interface Seller {
...
+      return
+    }
+
+    const isThisItem = e.data.items[0].id?.toString() === sku?.itemId
+
+    if (!isThisItem) {
+      return
+    }
+
+    window.location.href = '/checkout/'
```
Comment (tlgimenes) [inline packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts:47]: we should add this navigation in the next frame so we are sure all events are registered

**Review by tlgimenes — CHANGES_REQUESTED** [review]

**Review by tlgimenes — APPROVED** [review]
LGTM

**Pull request merged**